pandas/core/window.py

```diff
@@ -847,10 +972,77 @@ class Rolling(_Rolling_and_Expanding):
     frequency by resampling the data. This is done with the default parameters
     of :meth:`~pandas.Series.resample` (i.e. using the `mean`).
     """
+    _attributes = _Window._attributes + ['_on']
+
+    def __init__(self, obj, window=None, min_periods=None, freq=None,
+                 center=False, win_type=None, on=None, axis=0, **kwargs):
+
+        self._on = kwargs.pop('_on', None)
+        self.on = on
+        super(Rolling, self).__init__(obj=obj, window=window,
+                                      min_periods=min_periods,
+                                      freq=freq, center=center,
+                                      win_type=win_type,
+                                      axis=axis, **kwargs)
+
+    @cache_readonly
+    def is_datetimelike(self):
+        return isinstance(self._on,
+                          (ABCDatetimeIndex,
+                           ABCTimedeltaIndex,
+                           ABCPeriodIndex))
 
     def validate(self):
         super(Rolling, self).validate()
-        if not com.is_integer(self.window):
+
+        # validate on
+        self.on, on = None, self.on
+
+        if on is None:
+            if self._on is None:
+                self._on = self.obj.index
+        elif (isinstance(self.obj, ABCDataFrame) and
+              on in self.obj.columns):
+            self._on = pd.Index(self.obj[on])
+            self.obj = self.obj.drop(on, axis=1)
+        else:
+            raise ValueError("invalid on specified as {0}, "
+                             "must be a column (if DataFrame) "
+                             "or None".format(on))
+
+        # we allow rolling on a datetimelike index
+        if (self.is_datetimelike and
+                isinstance(self.window, (compat.string_types, DateOffset))):
+
+            # must be monotonic for on
+            if not self._on.is_monotonic:
+                formatted = on or 'index'
+                raise ValueError("The on specified: {0} must be "
```

Comment:
"the on specified" looks a bit strange here (certainly if you didn't specify it). Maybe just "{on} must be monotonic"?